test/ruby/test_allocation.rb

```diff
@@ -0,0 +1,608 @@
+# frozen_string_literal: false
+require 'test/unit'
+
+class TestAllocation < Test::Unit::TestCase
+  def check_allocations(num_arrays, num_hashes, setup_code, check_code)
+    assert_separately([], <<~BODY)
+      def self.num_allocations(ary, counts)
+        ObjectSpace.count_objects(counts)
+        arrays = counts[:T_ARRAY]
+        hashes = counts[:T_HASH]
+        yield
+        ObjectSpace.count_objects(counts)
+        arrays -= counts[:T_ARRAY]
+        hashes -= counts[:T_HASH]
+        ary[0] = -arrays
+        ary[1] = -hashes
+        ary
+      end
+
+      GC.start
+      GC.disable
+
+      allocations = [0, 0]
+      counts = {}
+      empty_array = empty_array = []
+      empty_hash = empty_hash = {}
+      array1 = array1 = [1]
+      hash1 = hash1 = {a: 2}
+      nill = nill = nil
+      block = block = lambda{}
```

Comment:
Not sure why these are assigned to twice.

Reply:
To avoid verbose warnings for the calls that do not use them.